clients/hadoopfs/src/test/java/io/lakefs/committer/LakeFSOutputCommitterTest.java

```diff
@@ -0,0 +1,155 @@
+
+        ClientConfiguration clientConfiguration = new ClientConfiguration()
+                .withSignerOverride("AWSS3V4SignerType");
+        String s3Endpoint = String.format("http://s3.local.lakefs.io:%d", s3.getMappedPort(9000));
+
+        s3Client = new AmazonS3Client(creds, clientConfiguration);
+
+        S3ClientOptions s3ClientOptions = new S3ClientOptions()
+            .withPathStyleAccess(true);
+        s3Client.setS3ClientOptions(s3ClientOptions);
+        s3Client.setEndpoint(s3Endpoint);
+
+        s3Bucket = "example";
+        s3Base = String.format("s3://%s", s3Bucket);
+        CreateBucketRequest cbr = new CreateBucketRequest(s3Bucket);
+        s3Client.createBucket(cbr);
+
+        SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("Spark test")
+                .set("spark.sql.shuffle.partitions", "17")
+                .set("spark.hadoop.mapred.output.committer.class","io.lakefs.committer.DummyOutputCommitter")
+                .set("spark.hadoop.fs.lakefs.impl","io.lakefs.LakeFSFileSystem")
+                .set("fs.lakefs.access.key", "AKIAIOSFODNN7EXAMPLE")
+                .set("fs.lakefs.secret.key", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY")
+                .set("fs.lakefs.endpoint", "http://localhost:8000/api/v1")
+                .set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+                .set(org.apache.hadoop.fs.s3a.Constants.ACCESS_KEY, S3_ACCESS_KEY_ID)
+                .set(org.apache.hadoop.fs.s3a.Constants.SECRET_KEY, S3_SECRET_ACCESS_KEY)
+                .set(org.apache.hadoop.fs.s3a.Constants.ENDPOINT, s3Endpoint);
+        this.spark = SparkSession.builder().config(sparkConf).getOrCreate();
+    }
+
+
+    @Test
+    public void testWriteTextfile() throws ApiException, IOException {
+        String output = "lakefs://example/main/some_df";
+
+        StructType structType = new StructType();
+        structType = structType.add("A", DataTypes.StringType, false);
+
+        List<Row> nums = new ArrayList<Row>();
+        nums.add(RowFactory.create("value1"));
+        nums.add(RowFactory.create("value2"));
+        Dataset<Row> df = spark.createDataFrame(nums, structType);
+
+        ArrayList<String> failed = new ArrayList<String>();
+
+        // TODO(Lynn): Add relevant file formats
+        ArrayList<String> fileFormats = new ArrayList<String>(Arrays.asList("text"));
+        for (String fileFormat : fileFormats) {
+            try {
+                // save data in selected format
+                String outputPath = String.format("%s.%s", output, fileFormat);
+                df.write().format(fileFormat).save(outputPath);
+
+                // read the data we just wrote
+                Dataset<Row> df_read = spark.read().format(fileFormat).load(outputPath);
+                Dataset<Row> diff = df.exceptAll(df_read);
+                Assert.assertTrue(diff.isEmpty());
```

Comment:
This tests one direction: that all rows written were actually in the input.  It is also interesting to verify that all rows in the input were actually written.  This code should do that too:
```suggestion
                Dataset<Row> diff = df.exceptAll(df_read);
                Assert.assertTrue("all rows written were read", diff.isEmpty());
                diff = df_read.exceptAll(df);
                Assert.assertTrue("all rows read were written", diff.isEmpty());
```
We might also stringify the first 20 rows of diff, so that if it fails we have some examples.  Something like this (but I did not test that it works!):
```suggestion
                Dataset<Row> diff = df.exceptAll(df_read);
                Assert.assertTrue(String.format("expected all rows written were read but go %s", diff.take(20).collect), diff.isEmpty());
                diff = df_read.exceptAll(df);
                Assert.assertTrue("...same...", diff.isEmpty());
```